app/src/test/java/ar/com/wolox/android/example/ui/auth/login/AuthPresenterTest.kt

```diff
@@ -0,0 +1,115 @@
+package ar.com.wolox.android.example.ui.auth.login
+
+import ar.com.wolox.android.example.model.LoginResponse
+import ar.com.wolox.android.example.network.repository.LoginRepository
+import ar.com.wolox.android.example.utils.Extras
+import ar.com.wolox.android.example.utils.UserSession
+import ar.com.wolox.wolmo.core.tests.CoroutineTestRule
+import ar.com.wolox.wolmo.core.tests.WolmoPresenterTest
+import ar.com.wolox.wolmo.networking.retrofit.handler.NetworkResponse
+import kotlinx.coroutines.runBlocking
+import org.junit.Rule
+import org.junit.Test
+import org.mockito.Mock
+import org.mockito.Mockito.mock
+import retrofit2.Response
+import com.nhaarman.mockitokotlin2.times
+import com.nhaarman.mockitokotlin2.verify
+import com.nhaarman.mockitokotlin2.whenever
+import com.nhaarman.mockitokotlin2.doReturn
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+
+@ExperimentalCoroutinesApi
+class AuthPresenterTest : WolmoPresenterTest<AuthView, AuthPresenter>() {
+
+    @get:Rule
+    val coroutineTestRule = CoroutineTestRule(runOnAllTests = true)
+
+    @Mock
+    lateinit var userSession: UserSession
+
+    @Mock
+    lateinit var loginRepository: LoginRepository
+
+    override fun getPresenterInstance() = AuthPresenter(loginRepository, userSession)
+
+    @Test
+    fun `When the email is empty`() = runBlocking {
+        presenter.onLoginButtonClicked("", Extras.MockTesting.PASSWORD).join()
+        val errors: List<LoginFormErrors> = listOf(
+            LoginFormErrors.INVALID_EMAIL,
+            LoginFormErrors.EMPTY_EMAIL
+        )
+
+        verify(view, times(1)).setErrors(errors)
+    }
+
+    @Test
+    fun `When the email is invalid`() = runBlocking {
+        presenter.onLoginButtonClicked(Extras.MockTesting.INVALID_EMAIL, Extras.MockTesting.PASSWORD).join()
+        val errors: List<LoginFormErrors> = listOf(LoginFormErrors.INVALID_EMAIL)
+
+        verify(view, times(1)).setErrors(errors)
+    }
+
+    @Test
+    fun `When the password is empty`() = runBlocking {
+        presenter.onLoginButtonClicked(Extras.MockTesting.EMAIL, "").join()
+        val errors: List<LoginFormErrors> = listOf(LoginFormErrors.EMPTY_PASSWORD)
+
+        verify(view, times(1)).setErrors(errors)
+    }
+
```

Comment:
Maybe you could add two tests: one for when the password is invalid and one in which both email and password are invalid